docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+# DCOS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DCOS, we prefer to keep Google Python Style Guide Revision 2.59 as the base and **extend** it with
+some changes that will make DCOS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we can use tools available in the ecosystem to our advantage, enforce consistency in the codebase, creating and publishing automatically generated documentation for our code.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR, discussing the merits or demerits to arrive at a consensus.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59](https://google.github.io/styleguide/pyguide.html)
+
+#### Summary of Style Requirements DCOS which extend Base Style Guide and are specific to DCOS.
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
```

Comment:
@jsrodman , smallest `__init__.py` is difficult to define.  The suggestion is to have empty `__init__.py` and not have function definitions or write code in `__init__.py`

* Use `__init__.py` only to represent that the dir is a python package. Have the definitions in the submodule.

Some argue that when you declare/include `ClassObj` from `submodule1 ` inside `pkg1/__init__.py` you could write it like this.

``` from pkg import ClassObj```

Instead of

```from pkg1.submodule1 import ClassObj```

* The point of the rule is, while former might be easier to read, it is ambiguous on where something is defined. And if you are using tools to navigate to definitions of the object, then former will lead to `__init__.py` and then, you have to search go to `submodule1` for actual definition. It becomes confusing for the minor benefit the re-declaration provides.